tools/codegen/src/generator/languages/rust_lazy_reader/generator.rs

```diff
@@ -0,0 +1,718 @@
+                    #( #verify_fields )*
+                    Ok(())
+                }
+            }
+        };
+        writeln!(output, "{}", q)?;
+        Ok(())
+    }
+}
+
+// in FixVec, all item size is same and known already, the count is unknown.
+impl LazyReaderGenerator for ast::FixVec {
+    fn gen_rust<W: io::Write>(&self, output: &mut W) -> io::Result<()> {
+        let (transformed_name, tc) = get_rust_type_category(self.item().typ());
+        generate_rust_common_array(
+            output,
+            self.name(),
+            transformed_name,
+            tc,
+            None,
+            Some(self),
+            None,
+        )?;
+
+        let name = ident_new(self.name());
+        let item_size = self.item_size();
+
+        let q = quote! {
+            impl #name {
+                pub fn verify(&self, _compatible: bool) -> Result<(), Error> {
+                    self.cursor.verify_fixvec(#item_size)?;
+                    Ok(())
+                }
+            }
+        };
+        writeln!(output, "{}", q)?;
+        Ok(())
+    }
+}
+
+impl LazyReaderGenerator for ast::DynVec {
+    fn gen_rust<W: io::Write>(&self, output: &mut W) -> io::Result<()> {
+        let (transformed_name, tc) = get_rust_type_category(self.item().typ());
+        generate_rust_common_array(
+            output,
+            self.name(),
+            transformed_name,
+            tc,
+            None,
+            None,
+            Some(self),
+        )?;
+
+        let verify_sub = {
+            let func = verify_typ(self.item().typ().as_ref(), quote!(self.get(i)?));
+            if func.is_empty() {
+                quote!()
+            } else {
+                quote!(for i in 0..self.len()? {
+                    #func
+                })
+            }
+        };
+        let val_compatible = if verify_sub.is_empty() {
+            quote!(_compatible)
+        } else {
+            quote!(compatible)
+        };
+
+        let name = ident_new(self.name());
+        let q = quote! {
+            impl #name {
+                pub fn verify(&self, #val_compatible: bool) -> Result<(), Error> {
+                    self.cursor.verify_dynvec()?;
+                    #verify_sub;
+                    Ok(())
+                }
+            }
+        };
+        writeln!(output, "{}", q)?;
+        Ok(())
+    }
+}
+
+impl LazyReaderGenerator for ast::Table {
+    fn gen_rust<W: io::Write>(&self, output: &mut W) -> io::Result<()> {
+        generate_rust_common_table(output, self.name(), self.fields(), None)?;
+        let field_count = self.fields().len();
+        let name = ident_new(self.name());
+
+        let verify_fields = self
+            .fields()
+            .iter()
+            .enumerate()
+            .map(|(_i, f)| verify_filed(f));
+
+        let q = quote! {
+            impl #name {
+                pub fn verify(&self, compatible: bool) -> Result<(), Error> {
```

Comment:
Personally, in a newly designed code generator, I would vote against introducing `compatible` field here. `compatible`, to me, is really a feature in molecule that we haven't thought through before. Typically what we need is `compatible` to work for one or two levels deep, but instead here we can only either disable compatibility completely, or have recursive compatibility. So to me, I personally recommend removing this feature for now in the generated code, till a later time where we understand better how we want to use this feature.